js/add-to-calendar.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @file JS file for Add to Calendar.
+ */
+
+(function localgovAddToCalendarScript(Drupal) {
+  Drupal.behaviors.localgovAddToCalendar = {
+    attach(context) {
+      const addToCalendarButtons = once('all-add-to-calendar-buttons', '.localgov-add-to-calendar__trigger', context);
+      if (addToCalendarButtons.length) {
+        addToCalendarButtons.forEach(button => {
+          const calendarModal = button.nextElementSibling;
+          const calendarModalClose = calendarModal.querySelector('.localgov-add-to-calendar__dialog-close-button');
+          button.addEventListener('click', () => {
+            calendarModal.showModal();
+          });
+          calendarModalClose.addEventListener('click', () => {
+            calendarModal.close();
```

Comment:
If we're launching a modal we'll need to pay attention to two things: focus and close actions.
We'll need to ensure that focus is in a sensible place in the modal, and then on close, that it goes back to a sensible place. Maybe back to the button that launched it?

For closing, we need to make sure that a modal can be closed by

- clicking a close button (which must be accessible by the keyboard, and might make a good sensible focus starting point?)
- clicking outside the modal
- hitting `esc`

And usually we'll want to do something to prevent page scroll whilst the modal is open, coz it's just weird.
Has LGD not already got any modal constructions in any of the other myriad modules we can lean on, which might have these things already addressed?